### ISSUE TITLE
3.0.0 Native crash

### ISSUE THREAD

**Issue by DummyCo** [twilio/twilio-chat-demo-android#42]
### Description

Got new heisenbug on 3.0.0
Not sure if that stacktrace is useful but whatever posting it here:

```
2018-04-10 12:10:41.312 20710-20710/? A/DEBUG: backtrace:
        #00 pc 00000000004bbbfc  /data/app/com-g-d1X2uqtQ8tZPpxsZaltA==/lib/arm64/libtwilio-rtd-native.so (rtd::TSEntityCache<long long>::getAdjacentInternal(long long const&, rtd::TSQueryOrder)+296)
        #01 pc 00000000004bbd4c  /data/app/com-g-d1X2uqtQ8tZPpxsZaltA==/lib/arm64/libtwilio-rtd-native.so (rtd::TSEntityCache<long long>::getFirstItem(rtd::TSQueryOrder)+280)
        #02 pc 00000000004b1394  /data/app/com-g-d1X2uqtQ8tZPpxsZaltA==/lib/arm64/libtwilio-rtd-native.so (rtd::TSList::resolveNextItem(rtd::TSItemId, rtd::TSQueryOrder, bool)+368)
        #03 pc 00000000004b18cc  /data/app/com-g-d1X2uqtQ8tZPpxsZaltA==/lib/arm64/libtwilio-rtd-native.so (rtd::TSList::queryPreviousItem(rtd::TSItemId, bool)+60)
        #04 pc 00000000004c1050  /data/app/com-g-d1X2uqtQ8tZPpxsZaltA==/lib/arm64/libtwilio-rtd-native.so (rtd::TSPaginator<std::__1::shared_ptr<rtd::ITSListItem const>>::chainedPopulator(std::__1::shared_ptr<rtd::future_impl::promise<rtd::TSOperationResult<std::__1::shared_ptr<rtd::ITSPaginator<std::__1::shared_ptr<rtd::ITSListItem const>>>>>>, std::__1::shared_ptr<rtd::TSPaginator<std::__1::shared_ptr<rtd::ITSListItem const>>>, rtd::TSItemId, bool)+172)
        #05 pc 00000000004c14a8  /data/app/com-g-d1X2uqtQ8tZPpxsZaltA==/lib/arm64/libtwilio-rtd-native.so (rtd::TSPaginator<std::__1::shared_ptr<rtd::ITSListItem const>>::createAsyncInternal(rtd::TSQueryOrder, rtd::TSQueryOrder, rtd::TSItemOrder, rtd::TSItemId, unsigned long, std::__1::shared_ptr<rtd::ITSItemQueryProvider<std::__1::shared_ptr<rtd::ITSListItem const>>>, bool)+460)
        #06 pc 00000000004b3770  /data/app/com-g-d1X2uqtQ8tZPpxsZaltA==/lib/arm64/libtwilio-rtd-native.so (rtd::TSList::queryItems(rtd::ITSQueryOptions<long long> const&) const+340)
        #07 pc 0000000000435cc4  /data/app/com-g-d1X2uqtQ8tZPpxsZaltA==/lib/arm64/libtwilio-rtd-native.so (rtd::TMDataList::query(rtd::TSQueryOptions<long long>, rtd::TMDataListQueryType, int)+200)
        #08 pc 0000000000435fec  /data/app/com-g-d1X2uqtQ8tZPpxsZaltA==/lib/arm64/libtwilio-rtd-native.so (rtd::TMDataList::queryLastItems(unsigned int, int)+204)
        #09 pc 000000000041efe8  /data/app/com-g-d1X2uqtQ8tZPpxsZaltA==/lib/arm64/libtwilio-rtd-native.so (rtd::TMessages::getLastMessages(std::__1::function<void (std::__1::shared_ptr<rtd::ITMResult>, std::__1::list<std::__1::shared_ptr<rtd::ITMessage>, std::__1::allocator<std::__1::shared_ptr<rtd::ITMessage>>>)>, unsigned int)+196)
        #10 pc 00000000003ce6cc  /data/app/com-g-d1X2uqtQ8tZPpxsZaltA==/lib/arm64/libtwilio-rtd-native.so (Java_com_twilio_chat_Messages_nativeGetLastMessages+412)
        #11 pc 00000000000422a0  /data/app/com-g-d1X2uqtQ8tZPpxsZaltA==/oat/arm64/base.odex (offset 0x3c000)
```

Unfortunately no stable reproduction :(

#### Chat Android SDK

3.0.0

#### Android API

27

#### Android Device

Nexus 5X

**Comment by berkus** [comment]
Thanks for report, this may be due to updated native libs; will investigate.

**Comment by DummyCo** [comment]
Some more info:
Looks like somewhat related to deleting the only one left message in channel.

```
2018-04-12 12:34:04.811 26282-26282/? A/DEBUG: *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
    Build fingerprint: 'google/bullhead/bullhead:8.1.0/OPM2.171019.029/4657601:user/release-keys'
    Revision: 'rev_1.0'
    ABI: 'arm64'
    pid: 25047, tid: 25047, name: m.example.application  >>> com.example.application <<<
    signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x1
    Cause: null pointer dereference
        x0   0000007e1db57480  x1   0000007e214b24b8  x2   0000000000000001  x3   0000007e1db57480
        x4   0000007e1db57480  x5   0000000000000000  x6   0000000000000001  x7   0000007e233c3900
        x8   0000007fc8313290  x9   0000000000000001  x10  0000000000000000  x11  0000000000000000
        x12  0000007e4120d8b8  x13  6163393830653439  x14  0000000000000000  x15  0000000000000002
        x16  0000007e233e78d0  x17  0000007ec3f9171c  x18  00000000147de088  x19  0000007fc8313290
        x20  0000007e214b2478  x21  0000007e214b24b8  x22  0000000000000001  x23  0000007fc83133a8
        x24  0000007fc8313558  x25  0000007fc83133c0  x26  0000007fc83133a0  x27  0000007fc83134a0
        x28  0000007e280ea218  x29  0000007fc83131b0  x30  0000007e22ec9b0c
        sp   0000007fc83131b0  pc   0000007e22ec9bfc  pstate 0000000060000000
2018-04-12 12:34:04.819 26282-26282/? A/DEBUG: backtrace:
        #00 pc 00000000004bbbfc  /data/app/com.example.application-iHX_j0xG9wq8NWCSalTNoQ==/lib/arm64/libtwilio-rtd-native.so (rtd::TSEntityCache<long long>::getAdjacentInternal(long long const&, rtd::TSQueryOrder)+296)
        #01 pc 00000000004bbd4c  /data/app/com.example.application-iHX_j0xG9wq8NWCSalTNoQ==/lib/arm64/libtwilio-rtd-native.so (rtd::TSEntityCache<long long>::getFirstItem(rtd::TSQueryOrder)+280)
2018-04-12 12:34:04.820 26282-26282/? A/DEBUG:     #02 pc 00000000004b1394  /data/app/com.example.application-iHX_j0xG9wq8NWCSalTNoQ==/lib/arm64/libtwilio-rtd-native.so (rtd::TSList::resolveNextItem(rtd::TSItemId, rtd::TSQueryOrder, bool)+368)
        #03 pc 00000000004b18cc  /data/app/com.example.application-iHX_j0xG9wq8NWCSalTNoQ==/lib/arm64/libtwilio-rtd-native.so (rtd::TSList::queryPreviousItem(rtd::TSItemId, bool)+60)
        #04 pc 00000000004c1050  /data/app/com.example.application-iHX_j0xG9wq8NWCSalTNoQ==/lib/arm64/libtwilio-rtd-native.so (rtd::TSPaginator<std::__1::shared_ptr<rtd::ITSListItem const>>::chainedPopulator(std::__1::shared_ptr<rtd::future_impl::promise<rtd::TSOperationResult<std::__1::shared_ptr<rtd::ITSPaginator<std::__1::shared_ptr<rtd::ITSListItem const>>>>>>, std::__1::shared_ptr<rtd::TSPaginator<std::__1::shared_ptr<rtd::ITSListItem const>>>, rtd::TSItemId, bool)+172)
        #05 pc 00000000004c14a8  /data/app/com.example.application-iHX_j0xG9wq8NWCSalTNoQ==/lib/arm64/libtwilio-rtd-native.so (rtd::TSPaginator<std::__1::shared_ptr<rtd::ITSListItem const>>::createAsyncInternal(rtd::TSQueryOrder, rtd::TSQueryOrder, rtd::TSItemOrder, rtd::TSItemId, unsigned long, std::__1::shared_ptr<rtd::ITSItemQueryProvider<std::__1::shared_ptr<rtd::ITSListItem const>>>, bool)+460)
        #06 pc 00000000004b3770  /data/app/com.example.application-iHX_j0xG9wq8NWCSalTNoQ==/lib/arm64/libtwilio-rtd-native.so (rtd::TSList::queryItems(rtd::ITSQueryOptions<long long> const&) const+340)
        #07 pc 0000000000435cc4  /data/app/com.example.application-iHX_j0xG9wq8NWCSalTNoQ==/lib/arm64/libtwilio-rtd-native.so (rtd::TMDataList::query(rtd::TSQueryOptions<long long>, rtd::TMDataListQueryType, int)+200)
        #08 pc 0000000000435fec  /data/app/com.example.application-iHX_j0xG9wq8NWCSalTNoQ==/lib/arm64/libtwilio-rtd-native.so (rtd::TMDataList::queryLastItems(unsigned int, int)+204)
        #09 pc 000000000041efe8  /data/app/com.example.application-iHX_j0xG9wq8NWCSalTNoQ==/lib/arm64/libtwilio-rtd-native.so (rtd::TMessages::getLastMessages(std::__1::function<void (std::__1::shared_ptr<rtd::ITMResult>, std::__1::list<std::__1::shared_ptr<rtd::ITMessage>, std::__1::allocator<std::__1::shared_ptr<rtd::ITMessage>>>)>, unsigned int)+196)
        #10 pc 00000000003cf600  /data/app/com.example.application-iHX_j0xG9wq8NWCSalTNoQ==/lib/arm64/libtwilio-rtd-native.so (Java_com_twilio_chat_Messages_setAllMessagesConsumedWithResult+552)
        #11 pc 0000000000042b98  /data/app/com.example.application-iHX_j0xG9wq8NWCSalTNoQ==/oat/arm64/base.odex (offset 0x3c000)
```

**Comment by berkus** [comment]
Thanks, I will investigate this scenario.

**Comment by tanmay-gp91** [comment]
Any update on this? It started crashing for me all of a sudden as well!

**Comment by berkus** [comment]
@tanmay-gp91 hmm, that's sad. Do you have any logs to add? It would be nice to have more detailed backtraces as well as logs. I will take a thorough look and try to reproduce it next week.

**Comment by berkus** [comment]
Reproduced this.

**Comment by berkus** [comment]
Bug is found and fixed, will appear in release after next (3.0.1 is ready to be out, so the next version after it should have the fix).

**Comment by berkus** [comment]
3.1.0 should have this
